feat(package): add function to check packages

italopessoa opened the pull request:
add function to check if the user has a specific package installed
still working on the message
closes #4

italopessoa committed:
feat(package): add function to check packages

add function to check if the user has a specific package installed
still working on the message

paulonotz0r reviewed (commented):
Not working on Windows. With `create-react-app` not installed, the following message is throwed:

`'create-react-app' is not recognized as an internal or external command,
operable program or batch file.
(node:10100) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: spawn create-react-app ENOENT`

jouderianjr commented on awesomeCreateApp.js:
```diff
@@ -15,8 +15,12 @@ const terminalOpts = {
+    spawn('create-react-app', [folderName], terminalOpts)
     .then( () => showQuestions(folderName))
+    //TODO ask for install the package
+  ).catch(() => console.log(`${emoji.get('scream_cat')} create-react-app is not installed`))
```
I think we could improve this message, maybe it's better explain for the user how the `create-react-app` could be installed. Also, could you add some colors (Using chalk)? for example:

```
${emoji.get('scream_cat')} ${chalk.red('Error: ')} create-react-app is not installed
Install create-react-app using: npm i -g create-react-app
```

italopessoa replied:
Is it the time to use `showCreateAppError()` ?

jouderianjr commented on awesomeCreateApp.js:
```diff
@@ -15,8 +15,12 @@ const terminalOpts = {
 const createApp = folderName => {
-  spawn('create-react-app', [folderName], terminalOpts)
+  checkPackageInstalled('create-react-app')
+  .then(() => 
```
Could you extract this:

```
spawn('create-react-app', [folderName], terminalOpts)
      .then( () => showQuestions(folderName))
```

create a new function ( could be called executeCreateReactApp os something related)

Wha do you think about? 😄

italopessoa replied:
Should it be an one line function as well?

jouderianjr commented:
Thank you for this PR @italopessoa ❤️

jouderianjr commented on awesomeCreateApp.js:
```diff
@@ -47,4 +51,8 @@ const showEndProcessText = () => {
   Promise.resolve()
 }
 
+const checkPackageInstalled = packageName => {
```
1 - This could be an one line function

2 - this could return error when the create-react-app is installed, so should be better use `create-react-app --version` 💃

italopessoa committed:
feat(package): refact function to check package #4

refactored
 - checkPackageInstalled: one line function
 - showCreateAppError: add `error` argument

created
 - executeCreateReactApp

jouderianjr commented on awesomeCreateApp.js:
```diff
@@ -13,15 +13,20 @@ const terminalOpts = {
 const showQuestions = folderName =>
   inquirer.prompt(questions).then(handleOptions.bind(this,folderName))
 
 // TODO: Add error message
```
Could you please remove this comment line ? 😄  then you can merge it!

italopessoa committed:
feat(package): remove TODO comment #4

italopessoa commented:
comment removed 🤙

italopessoa committed:
feat(package): add little tab back :art:

italopessoa commented:
Hey @paulonotz0r the not **recognized error** was fixed here and the `UnhandledPromiseRejectionWarning` message is being tracked here #11

jvcjunior reviewed (approved):
Awesome!
